sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -121,3 +158,70 @@ func checkHTTPSForAuth(req *policy.Request, allowHTTP bool) error {
 	}
 	return nil
 }
+
+// parseCAEChallenge returns a *authChallenge representing Response's CAE challenge (nil when Response has none).
+// If Response includes a CAE challenge having invalid claims, it returns a NonRetriableError.
+func parseCAEChallenge(res *http.Response) (*authChallenge, error) {
+	var (
+		caeChallenge *authChallenge
+		err          error
+	)
+	for _, c := range parseChallenges(res) {
+		if c.scheme == "Bearer" {
+			if claims := c.params["claims"]; claims != "" && c.params["error"] == "insufficient_claims" {
+				b, de := base64.StdEncoding.DecodeString(claims)
+				if de != nil {
+					// don't include the decoding error because it's something
+					// unhelpful like "illegal base64 data at input byte 42"
+					err = errorinfo.NonRetriableError(errors.New("authentication challenge contains invalid claims: " + claims))
+					break
+				}
+				c.params["claims"] = string(b)
```

Comment:
Since we're ranging over `parseChallenges` it seems like we could hit this case multiple times (in theory anyways), so the last one wins. Is the ok?

Reply:
Yes, "one response having multiple Bearer challenges" isn't a real scenario. There could in principle be another challenge, for a different scheme, so the loop could continue after finding a Bearer CAE challenge. However, this function wouldn't do anything with those challenges. I suppose it may as well `break` after finding the CAE challenge though.

Reply:
I like the idea of adding a break as it makes it clear that we expect only one challenge and accept the first one we find.